grafana-plugin/src/state/plugin/index.ts

```diff
@@ -0,0 +1,273 @@
+import { getBackendSrv } from '@grafana/runtime';
```

Comment:
there was a lot of duplicated logic/function calls between `rootBaseStore` and `src/PluginConfigPage/PluginConfigPage.tsx`. This has all been centralized/tested here. I also decided to make these methods as `static` methods on an ES6 class (this made mocking/testing intra-module calls way easier; alternative was dependency-injection or some other dirty hacks)